hail/python/hailtop/config/user_config.py

```diff
@@ -117,3 +120,146 @@ def get_remote_tmpdir(caller_name: str,
+        warnings.warn(
+            f'When reading GCS requester pays configuration from '
+            f'spark-defaults.conf ({spark_conf.conf_path}) Hail treats the mode '
+            f'ENABLED as AUTO.'
+        )
+
+    if spark_conf.buckets is not None:
+        warnings.warn(
+            f'When reading GCS requester pays configuration from '
+            f'spark-defaults.conf ({spark_conf.conf_path}) with mode '
+            f'{spark_conf.mode}, found buckets: {spark_conf.buckets}.'
+            f'The buckets are ignored in this mode.'
+        )
+
+    return spark_conf.project
+
+
+class SparkConfGcsRequesterPaysMode(Enum):
+    ENABLED = 'ENABLED'
+    DISABLED = 'DISABLED'
+    AUTO = 'AUTO'
+    CUSTOM = 'CUSTOM'
+
+
+@dataclass
+class SparkConfGcsRequseterPaysConfiguration:
+    mode: Optional[SparkConfGcsRequesterPaysMode]
+    project: Optional[str]
+    buckets: Optional[List[str]]
+    conf_path: str
+
+
+def spark_conf_path() -> Optional[str]:
+    try:
+        from pyspark.find_spark_home import _find_spark_home  # pylint: disable=import-outside-toplevel
+    except ImportError:
+        return None
+    return os.path.join(_find_spark_home(), 'conf', 'spark-defaults.conf')
+
+
+def get_spark_conf_gcs_requester_pays_configuration() -> Optional[SparkConfGcsRequseterPaysConfiguration]:
+    mode: Optional[SparkConfGcsRequesterPaysMode] = None
+    project: Optional[str] = None
+    buckets: Optional[List[str]] = None
+    path = spark_conf_path()
+    if path is not None and os.path.exists(path):
+        with open(path, encoding='utf-8') as f:
+            for line in f.readlines():
+                setting = line.rstrip('\n')
+                maybe_var_and_val = setting.split(' ')
+                if len(maybe_var_and_val) != 2:
+                    raise ValueError(f'Found spark-defaults.conf file line with more than one space: {line}')
+                var, val = maybe_var_and_val
+                if var == 'spark.hadoop.fs.gs.requester.pays.mode':
+                    if val == SparkConfGcsRequesterPaysMode.ENABLED:
+                        mode = SparkConfGcsRequesterPaysMode.ENABLED
+                    elif val == SparkConfGcsRequesterPaysMode.DISABLED:
+                        mode = SparkConfGcsRequesterPaysMode.DISABLED
+                    elif val == SparkConfGcsRequesterPaysMode.AUTO:
+                        mode = SparkConfGcsRequesterPaysMode.AUTO
+                    elif val == SparkConfGcsRequesterPaysMode.CUSTOM:
+                        mode = SparkConfGcsRequesterPaysMode.CUSTOM
+                    else:
+                        raise ValueError(
+                            f'When reading GCS requester pays configuration from '
+                            f'spark-defaults.conf ({path}) an unknown mode was '
+                            f'found: {val}. Expected ENABLED, AUTO, CUSTOM, or '
+                            f'DISABLED.'
+                        )
```

Comment:
I'm pretty sure you should just be able to do `mode = SparkConfGcsRequesterPaysMode(val)`